kustomize/commands/edit/fix/fix_test.go

```diff
@@ -96,32 +90,273 @@ patches:
+`,
+		},
+		{
+			name: "TestFixOutdatedPatchesStrategicMergeFieldTitle",
+			input: `
+patchesStrategicMerge:
+- |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: nginx
+  spec:
+    template:
+      spec:
+        containers:
+          - name: nginx
+            image: nignx:latest
+`,
+			expected: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+patches:
+- patch: |-
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: nginx
+    spec:
+      template:
+        spec:
+          containers:
+            - name: nginx
+              image: nignx:latest
+`,
+		},
+		{
+			name: "TestFixAndMergeOutdatedPatchesStrategicMergeFieldTitle",
+			input: `
+patchesStrategicMerge:
+- |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: nginx
+  spec:
+    template:
+      spec:
+        containers:
+          - name: nginx
+            image: nignx:latest
+patches:
+- path: patch2.yaml
+  target:
+    kind: Deployment
+`,
+			expected: `
+patches:
+- path: patch2.yaml
+  target:
+    kind: Deployment
+- patch: |-
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: nginx
+    spec:
+      template:
+        spec:
+          containers:
+            - name: nginx
+              image: nignx:latest
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+`,
+		},
+		{
+			name: "TestFixOutdatedPatchesStrategicMergeToPathFieldTitle",
+			input: `
+patchesStrategicMerge:
+- deploy.yaml
+`,
+			files: map[string]string{
+				"deploy.yaml": `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx
+spec:
+  template:
+    spec:
+      containers:
+        - name: nginx
+          env:
+            - name: CONFIG_FILE_PATH
+              value: home.yaml`,
+			},
+			expected: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+patches:
+- path: deploy.yaml
+`,
+		},
+		{
+			name: "TestFixOutdatedPatchesStrategicMergeToPathFieldYMLTitle",
+			input: `
+patchesStrategicMerge:
+- deploy.yml
+`,
+			files: map[string]string{
+				"deploy.yml": `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+name: nginx
+spec:
+template:
+spec:
+containers:
+  - name: nginx
+    env:
+      - name: CONFIG_FILE_PATH
+        value: home.yaml`,
+			},
+			expected: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+patches:
+- path: deploy.yml
+`,
+		},
+		{
+			name: "Test fix outdated patchesStrategicMerge from a file and one string literal",
+			input: `
+patchesStrategicMerge:
+- deploy.yaml
+- |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: nginx
+  spec:
+    template:
+      spec:
+        containers:
+          - name: nginx
+            env:
+              - name: CONFIG_FILE_PATH
+                value: home.yaml
+`,
+			files: map[string]string{
+				"deploy.yaml": `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+name: nginx
+spec:
+template:
+spec:
+containers:
```

Comment:
same here (and a few other places)

I know it doesn't really matter for these tests, but it's good to make sure our tests have valid input

Reply:
> I know it doesn't really matter for these tests, but it's good to make sure our tests have valid input

Sorry, I couldn't find that before committing.
I completely agree with your opinion.

I fixed everything.